_posts/2022-07-19-react-to-full-stack.md

```diff
@@ -0,0 +1,606 @@
+---
+layout: post
+title: React To Full Stack
+date: 2022-07-19
+tags: [react, mui, typescript, aws]
+author: Dale Murugan
+image: /img/fivetranslacklambdalogo.png
+description: "Level-up from a React to a Full Stack developer. A definitive guide for junior developers"
+---
+
+![React To Full Stack](/img/react-to-full-stack/react-to-full-stack-main.png)
+
+<br/>
+
+React seems to be the starting place for a lot of new developers, but what do you learn next? I know from personal experience that it can be daunting as a junior developer trying to decide what backend tech stack to go with. Should you use a SQL or NoSQL database? Should you use REST or GraphQL? How do you manage authentication? Should you learn .Net or Node? What should you use for hosting? What even is a Docker?
+
+This article will save you the decision fatigue. Assuming that you know a bit of React, you can follow this step-by-step guide to build and deploy a full stack cloud native application, using AWS's serverless infrastructure.
+
+---
+
+<br>
+## Overview
+ We are going to create the backend for a todo react app that I prepared earlier. The app is built with typescript and styled with material UI (MUI); we won't be covering either of these as there are plenty of useful free resources on these topics online. Instead, we will be focusing on creating the backend infrasture and hosting the application.
+
+As previously stated, our backend will be on the cloud, using AWS's serverless infrastructure. With AWS serverless infrastructure, building your backend is a lot easier and faster. There's less overhead to worry about like managing servers, system patching etc. It is also highly scalable, so you won't need to worry about adding extra resources as your app userbase grows. Additionally, it also has a pay-for-value billing model, which means you only pay for what you use. AWS also continues to be the largest cloud provider in the market.
+
+Let's start by defining some basic requirements for full stack CRUD (create, read, update, delete) app. Generally, a full stack CRUD application requires the following:
+
+- Ability to serve your application to users.
+- Ability to authenticate users to enter your app.
+- Somewhere to store your application build.
+- Ability to handle authorisation for API calls made from the app.
+- Ability to handle interactions between the app and database.
+- A database to store and retrieve information.
+
+**Basic Architecture Diagram**
+![Basic Architecture diagram](/img/react-to-full-stack/react-to-full-stack-basic-architecture.png)
```

Comment:
As per this Hot Tip, specify both the image dimension + lazy loading attributes to ALL of your "below the fold" images to improve the performance of your blog. 

https://mechanicalrock.slack.com/archives/CGVBXTHAP/p1657500766927879

Reply:
Ahh also good to know!